### PR TITLE
feat(wallet): Add Token Options Menu

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/23792

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
https://user-images.githubusercontent.com/48117473/185103808-0db7c9b0-3d47-4e01-ad5c-e24c73a61a6a.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/asset-more-popup/index.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/asset-more-popup/index.tsx:1]: Can we rename this file to `asset-more-popup.tsx`?

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/asset-more-popup/style.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/asset-more-popup/style.ts:1]: Can we rename this to `asset-more-popup.style.ts`?

File: components/brave_wallet_ui/stories/locale.ts [hunk 3]
```diff
@@ -668,5 +668,16 @@ provideStrings({
...
+  braveWalletPerksBrowserNative: 'Browser-native, no risky extensions',
+
+  // Portfolio Asset More Popup
+  braveWalletPortfolioTokenDetailsMenuLabel: 'Token details',
+  braveWalletPortfolioViewOnExplorerMenuLabel: 'View on block explorer',
+  braveWalletPortfolioHideTokenMenuLabel: 'Hide token',
+  braveWalletHideTokenModalTitle: 'Hide token',
+
+  // Token detail modals
+  braveWalletMakeTokenVisibleInstructions: 'You can make this asset visible in the future by going to "Visible assets" portfolio tab',
```
Comment (josheleonard) [inline components/brave_wallet_ui/stories/locale.ts:680]: Shouldn't this be something like 'You can make this asset visible again in the future by clicking the "+ Visible assets" button at the bottom of the "Portfolio" tab'

File: components/resources/wallet_strings.grdp [hunk 4]
```diff
@@ -577,4 +577,11 @@
...
   <message name="BRAVE_WALLET_UNLOCK_NEEDED_TO_CREATE_ACCOUNT" desc="Message to inform the user that their wallet must be unlocked before a new account can be created">Unlock needed to create an account</message>
   <message name="BRAVE_WALLET_CLICK_TO_SWITCH" desc="Action text for switching selection">Click to switch</message>
+  <message name="IDS_BRAVE_WALLET_PORTFOLIO_TOKEN_DETAILS_MENU_LABEL" desc="Label for token details menu item">Token details</message>
+  <message name="IDS_BRAVE_WALLET_PORTFOLIO_VIEW_ON_EXPLORER_MENU_LABEL" desc="Label for view on explorer menu item">View on block explorer</message>
+  <message name="IDS_BRAVE_WALLET_PORTFOLIO_VIEW_HIDE_TOKEN_LABEL" desc="Label for hide token menu item">Hide <ph name="SYMBOL"><ex>BAT</ex>$1</ph></message>
+  <message name="IDS_BRAVE_WALLET_HIDE_TOKEN_MODAL_TITLE" desc="Modal title for hide token modal">Hide token?</message>
+  <message name="IDS_BRAVE_WALLET_MAKE_TOKEN_VISIBLE_INSTRUCTIONS" desc="Instructions for making a token visible">You can make this asset visible in the future by going to "Visible assets" portfolio tab</message>
```
Comment (josheleonard) [inline components/resources/wallet_strings.grdp:584]: ditto

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/asset-more-popup/style.ts (hunk 2 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/asset-more-popup/style.ts]: Fixed here: f0f6f66ffb7984f3a43025eb001034ee77d394e8

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/asset-more-popup/index.tsx (hunk 1 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/asset-more-popup/index.tsx]: Fixed here: f0f6f66ffb7984f3a43025eb001034ee77d394e8

File: components/brave_wallet_ui/stories/locale.ts (hunk 3 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/stories/locale.ts]: Fixed here: e764aecb16be39852d678141a5b27c75933e04cd

File: components/resources/wallet_strings.grdp (hunk 4 above)
Comment (muliswilliam) [reply components/resources/wallet_strings.grdp]: Fixed here: e764aecb16be39852d678141a5b27c75933e04cd

**Review by josheleonard — APPROVED** [review]

**Pull request merged**